contrib/msggen/msggen/__main__.py

```diff
@@ -1,153 +1,31 @@
-    resp_file = base_path / f"{name.lower()}.schema.json"
-    request = CompositeField.from_js(json.load(open(req_file)), path=name)
-    response = CompositeField.from_js(json.load(open(resp_file)), path=name)
-
-    # Normalize the method request and response typename so they no
-    # longer conflict.
-    request.typename += "Request"
-    response.typename += "Response"
-
-    return Method(
-        name=method_name_override.get(name, name),
-        request=request,
-        response=response,
-    )
-
-
-def load_jsonrpc_service():
-    method_names = [
-        "Getinfo",
-        "ListPeers",
-        "ListFunds",
-        "SendPay",
-        "ListChannels",
-        "AddGossip",
-        "AutoCleanInvoice",
-        "CheckMessage",
-        "Close",
-        "Connect",
-        "CreateInvoice",
-        "Datastore",
-        "CreateOnion",
-        "DelDatastore",
-        "DelExpiredInvoice",
-        "DelInvoice",
-        "Invoice",
-        "ListDatastore",
-        "ListInvoices",
-        "SendOnion",
-        "ListSendPays",
-        "ListTransactions",
-        "Pay",
-        "ListNodes",
-        "WaitAnyInvoice",
-        "WaitInvoice",
-        "WaitSendPay",
-        "NewAddr",
-        "Withdraw",
-        "KeySend",
-        "FundPsbt",
-        "SendPsbt",
-        "SignPsbt",
-        "UtxoPsbt",
-        "TxDiscard",
-        "TxPrepare",
-        "TxSend",
-        # "decodepay",
-        # "decode",
-        # "delpay",
-        # "disableoffer",
-        "Disconnect",
-        "Feerates",
-        # "fetchinvoice",
-        # "fundchannel_cancel",
-        # "fundchannel_complete",
-        # "fundchannel",
-        # "fundchannel_start",
-        # "funderupdate",
-        # "getlog",
-        "GetRoute",
-        # "getsharedsecret",
-        "ListForwards",
-        # "listoffers",
-        "ListPays",
-        # "multifundchannel",
-        # "multiwithdraw",
-        # "offerout",
-        # "offer",
-        # "openchannel_abort",
-        # "openchannel_bump",
-        # "openchannel_init",
-        # "openchannel_signed",
-        # "openchannel_update",
-        # "parsefeerate",
-        "Ping",
-        # "plugin",
-        # "reserveinputs",
-        # "sendcustommsg",
-        # "sendinvoice",
-        # "sendonionmessage",
-        # "setchannelfee",
-        "SignMessage",
-        # "unreserveinputs",
-        # "waitblockheight",
-        # "ListConfigs",
-        # "check",  # No point in mapping this one
-        # "Stop",  # Breaks a core assumption (root is an object) can't map unless we change this
-        # "notifications",  # No point in mapping this
-        # "help",
-    ]
-    methods = [load_jsonrpc_method(name) for name in method_names]
-    service = Service(name="Node", methods=methods)
-    service.includes = ['primitives.proto']  # Make sure we have the primitives included.
-    return service
-
-
-def gengrpc(service, meta):
+def gengrpc(generator_chain: GeneratorChain, meta):
```

Comment:
Mh! would be cool, but I didn't change the name, let me ping @cdecker for this

Reply:
You didn't change the name, but you changed the behavior of the function. So what I meant is that you *should* change the name. Original name by @cdecker is fine in the original context but not anymore.

Reply:
well, the behavior of the function is not changed, the function configures the generator with the resulting path and the metadata.

But as you preferer, it is not important to me since it is an internal function!